admin/services/group.service.js

```diff
@@ -86,4 +86,19 @@ groupService.create = async (groupName, groupPupils, schoolId) => {
   return newGroup.insertId
 }
 
+/**
+ * Find groups that have pupils that can get PINs assigned.
+ * @param schoolId
+ * @param pupils
+ * @returns {Promise<*>}
+ */
+groupService.filterGroupsByPupil = async (schoolId, pupils) => {
+  let pupilIds = ''
+  if (pupils.length > 0) {
+    pupils.map(p => { if (p.group_id) pupilIds += "'" + p.group_id + "', " })
```

Comment:
i am abit confused with this if block.
What exactly is it supposed to be doing?

Reply:
If you follow-up the code you'll see that it concatenates values to be used in an IN statement in the data access service.

Reply:
It needs to use parameters on the insert if it's user-supplied data.  Please confirm?  You may find `sqlService.buildParameterList` list useful.

Reply:
@pris54 this work should be done in the data service.  Pass an array to the data service and build a set of parameters.  see other data services for reference.  passing a string to be used in a query is dangerous

Reply:
@pris54 see `checkWindowDataService.sqlFindCheckWindowsAssignedToForms` for a reference on how to do it correctly.  the `groupDataService.sqlFindGroupsByIds` method should never have made it to master.  we obviously didnt catch this in a previous review

Reply:
@GuyHarwood Changes done.